src/libraries/System.Private.CoreLib/src/System/Random.cs

```diff
@@ -218,7 +218,73 @@ public virtual int Next(int maxValue)
             return (int)(Sample() * maxValue);
         }
 
-        /*=====================================Next=====================================
+        private long FullLong()
+        {
+            long result = 0;
+
+            for (int i = 0; i < 4; i++)
+            {
+                short part;
+                unchecked
+                {
+                    part = (short) InternalSample();
+                }
+
+                result |= (long) part;
+                result <<= 16;
+            }
+
+            return result;
+        }
+
+        /*==================================NextInt64===================================
+        **Returns: An long [0..long.maxValue)
+        **Arguments: maxValue -- One more than the greatest legal return value.
+        **Exceptions: None.
+        ==============================================================================*/
+        public virtual long NextInt64()
+        {
+            return FullLong();
```

Comment:
See problems with `FullLong()`, but the current implementation does not obey the documented contract.